src/main/java/com/minecolonies/core/tileentities/TileEntityRack.java

```diff
@@ -178,6 +178,12 @@ public boolean hasItemStack(final ItemStack stack, final int count, final boolea
         return content.getOrDefault(checkItem, 0) >= count;
     }
 
+    @Override
+    public boolean hasItemStorage(final ItemStorage storage, final int count)
+    {
+        return content.getOrDefault(storage, 0) >= count;
```

Comment:
hm looking at this I think this actually will box the 0 to an Integer and back, incase its missing. (same as the hasItemStack method above actually) maybe we should just have a constant for a zero Integer

Reply:
FWIW, Java caches the 0 Integer already, and uses that every time it needs to box up 0